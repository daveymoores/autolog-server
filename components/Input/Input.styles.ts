import css from "styled-jsx/css";

export default css`
  input {
    height: var(--cellHeight);
    box-sizing: border-box;
    max-width: 380px;
    width: 100%;
    padding: 0;
    margin-bottom: 25px;
    text-align: center;
    border-radius: 0;
  }
`;
