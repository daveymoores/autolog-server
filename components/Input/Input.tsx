import React from "react";

import styles from "../Input/Input.styles";

const Input: React.FC = () => (
  <>
    <input type="text" className="t-gen--cell" />
    <style jsx>{styles}</style>
  </>
);

export default Input;
