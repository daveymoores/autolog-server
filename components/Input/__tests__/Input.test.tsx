import "@testing-library/jest-dom";

import { render } from "@testing-library/react";
import React from "react";

import Input from "../Input";

// Mock the styles to avoid styled-jsx issues
jest.mock("../Input.styles", () => ({
  __esModule: true,
  default: "",
}));

describe("Input Component", () => {
  it("renders an input element", () => {
    const { container } = render(<Input />);
    const inputElement = container.querySelector("input");
    expect(inputElement).toBeInTheDocument();
  });

  it("has type of text", () => {
    const { container } = render(<Input />);
    const inputElement = container.querySelector("input");
    expect(inputElement).toHaveAttribute("type", "text");
  });

  it("has the correct class name", () => {
    const { container } = render(<Input />);
    const inputElement = container.querySelector("input");
    expect(inputElement).toHaveClass("t-gen--cell");
  });
});
