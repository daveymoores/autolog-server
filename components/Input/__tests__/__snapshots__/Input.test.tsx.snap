// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders input unchanged 1`] = `
<input
  className="jsx-353473093 t-gen--cell"
  type="text"
/>
`;
